### ISSUE TITLE
Storage for contracts with the same name

### ISSUE THREAD

**Issue by DanglSan** [naddison36/sol2uml#92]
Running `sol2uml storage -c *same name*` on folder with files, containing contracts with the same name, will return storage diagram for contract with the smallest filename in sort order. 

Example:

`main.sol`:
```
pragma solidity ^0.8.7;

import {A as A1} from "./Lib1.sol";
import {A as A2} from "./Lib2.sol";

contract A is A1, Math2 {
    uint32 C1 = 0;
}
```

`Lib1.sol`:
```
pragma solidity ^0.8.7;

contract A {
    uint32 C2 = 0;
}
```

`Lib2.sol`:
```
pragma solidity ^0.8.7;

contract A {
    uint32 C3 = 0;
}
```

In this example storage will be generated for `A` from `Lib1.sol`. But if you rename `main.sol` in `amain.sol` result will be for `A` from `amain.sol`.

Maybe it can be fixed by choosing both filename and contract name?

**Comment by naddison36** [comment]
Thanks for raising this issue. I've added a `-cf, --contractFile <filename>` option for storage diagrams in v2.1.5.

The `contractFile` can just be a filename like `main.sol`. Or it can be a relative filename from the working folder. eg `src/contracts/main.sol`.

**Comment by DanglSan** [comment]
Are you sure, it works properly? Cause I just have checked and got an error: `Failed to find contract with name "A" in filename "./same_name/main.sol"`. Also tried other paths and contracts. I got this error on Windows and WSL.

**Comment by DanglSan** [comment]
UPD: it works properly for filename: `sol2uml storage ./same_name -c A -cf main.sol`. But I still have problems with relative paths.

**Comment by DanglSan** [comment]
I think, I got it. There are some problems with path comparison, so `sol2uml storage ./same_name -c A -cf same_name\main.sol` works fine, while all other options such as:
- `sol2uml storage ./same_name -c A -cf ./same_name/main.sol`
- `sol2uml storage ./same_name -c A -cf same_name/main.sol`
- `sol2uml storage ./same_name -c A -cf .\same_name\main.sol`

return an error.

**Comment by naddison36** [comment]
yeah, the relative paths has to be exact. Thinking about it some more, I can pass it through something like `path.normalize` to pick up your examples above.
I'll reopen this issue.

**Comment by naddison36** [comment]
I've done a change on a branch that will hopefully work for
- `sol2uml storage ./same_name -c A -cf .\same_name\main.sol`

I don't think the forward slashes will work on Windows.

You can test by installing the branch version with
```
npm i -g github:@naddison36/sol2uml#contract-filename
```

I can include this change in the next release.

**Comment by DanglSan** [comment]
I have tested the same example and it still doesn`t work. Interesting is that `sol2uml storage ./same_name -c A -cf .\main.sol` now works fine. I am not sure, but might be nessesary to add `path.normalize` in [this](https://github.com/naddison36/sol2uml/blob/b3254dc69ec77463920985a81589ff2b5ef411df/lib/converterClasses2Storage.js#L44) comparison?

**Comment by naddison36** [comment]
I've added the extra `path.normalize`. Can you try v2.1.6

**Comment by DanglSan** [comment]
Thx, in most situations it works great now, but when I use temporary directory like `C:\Users\user\AppData\Local\Temp\.tmpYLITLN\main.sol` for some reason `relative_path` [here ](https://github.com/naddison36/sol2uml/blob/51f7c141d7b8a07047226df52395cd3c72ac4c58/lib/converterClasses2Storage.js#L39) has value `..\..\..\AppData\Local\Temp\.tmpYLITLN\main.sol`. And comparison of `..\..\..\AppData\Local\Temp\.tmpYLITLN\main.sol` and `.tmpYLITLN\main.so` returns false, so I get `Failed to find contract with name` error again.

**Comment by RonnyGiron** [comment]
> Running `sol2uml storage -c *same name*` on folder with files, containing contracts with the same name, will return storage diagram for contract with the smallest filename in sort order.
> 
> Example:
> 
> `main.sol`:
> 
> ```
> pragma solidity ^0.8.7;
> 
> import {A as A1} from "./Lib1.sol";
> import {A as A2} from "./Lib2.sol";
> 
> contract A is A1, Math2 {
>     uint32 C1 = 0;
> }
> ```
> 
> `Lib1.sol`:
> 
> ```
> pragma solidity ^0.8.7;
> 
> contract A {
>     uint32 C2 = 0;
> }
> ```
> 
> `Lib2.sol`:
> 
> ```
> pragma solidity ^0.8.7;
> 
> contract A {
>     uint32 C3 = 0;
> }
> ```
> 
> In this example storage will be generated for `A` from `Lib1.sol`. But if you rename `main.sol` in `amain.sol` result will be for `A` from `amain.sol`.
> 
> Maybe it can be fixed by choosing both filename and contract name?

Please update all contracts with same name and contract owner
